src/host-config.ts

```diff
@@ -17,68 +17,52 @@ function handleErrorInNextTick(error: Error) {
   });
 }
 
-/** Drill down and skip any proxy nodes to get the _real_ one */
-function getFirstNonProxyNode(n: JsonNode): JsonNodeWithoutProxy | undefined {
-  if (n.type === "proxy") {
-    if (n.valueNode && n.valueNode?.type !== "proxy") {
-      return n.valueNode;
-    }
-
-    return undefined;
-  }
-
-  return n;
-}
+/** Validate that the given child can appear in the parent */
+function validateNesting(parent: JsonNode, child: JsonNode): void {}
 
 /** Append a child to a parent node */
 function appendChild(parent: JsonNode, child: JsonNode) {
   child.parent = parent;
-  const realChild = getFirstNonProxyNode(child);
-
-  if (realChild === undefined) {
-    return parent;
-  }
 
-  realChild.parent = parent;
+  validateNesting(parent, child);
 
   switch (parent.type) {
     case "array":
-      if (realChild.type === "property") {
-        throw new Error("Property cannot be a child of an array");
-      }
+      // if (realChild.type === "property") {
+      //   throw new Error("Property cannot be a child of an array");
+      // }
```

Comment:
Do these need to be added back?

Reply:
👍  Yeah i was in the process of moving it to a separate `validateNesting` handler that i still need to fill out